tensorflow-utils/src/main/java/org/tensorflow/util/buffer/impl/raw/UnsafeReference.java

```diff
@@ -0,0 +1,87 @@
+package org.tensorflow.util.buffer.impl.raw;
+
+import java.lang.reflect.Field;
+import sun.misc.Unsafe;
+
+/**
+ * Holds a valid reference to an instance of sun.misc.Unsafe.
+ */
+public final class UnsafeReference {
+
+  /**
+   * Creates a reference to {@code unsafe}
+   *
+   * The reference can be created only if the instance exposes the methods required by
+   * this library, otherwise an exception is thrown
+   *
+   * @param unsafe instance of {@code Unsafe}
+   * @return a valid reference to {@code unsafe}
+   * @throws IllegalArgumentException if {@code unsafe} instance does not meet requirements for this library
+   */
+  public static UnsafeReference from(Object unsafe) {
```

Comment:
Does this method need to be public? And can it take a stronger type than Object?

Reply:
yes, needs to be public, see my previous comment. The reason of accepting any `Object` here is to avoid uncaught `ClassNotFoundException` on a JDK that does not have `sun.misc.Unsafe`... but I just realized that I forgot that exception type in the `catch` clause below, I'll fix it :)